### PR TITLE
EE-805 update submission to deal with multi year and multi psubs

### PR DESCRIPTION
**Pull request by Pascoe121**

### CODE REVIEW

File: app/services/SubmissionService.scala [hunk 1]
```diff
@@ -53,26 +53,45 @@ class SubmissionService @Inject()(
...
-            val taxYear = TaxYearSelection.getTaxYear(taxYearSelection)
+
+        val psubsToUpdate: Seq[(Int, Seq[PSub])] = claimYears.flatMap {
+          year =>
+            subscriptions.get(getTaxYear(year)).filter(_.nonEmpty).map {
+              psubs =>
+                getTaxYear(year) -> psubs
+            }
+        }
+
```
Comment (JoeSimmonds) [inline app/services/SubmissionService.scala]: maybe 
`        val psubsToUpdate = for {
          year <- claimYears
          psubs <- subscriptions.get(getTaxYear(year)).filter(_.nonEmpty)
        } yield getTaxYear(year) -> psubs`

File: app/services/SubmissionService.scala [hunk 2]
```diff
@@ -53,26 +53,45 @@ class SubmissionService @Inject()(
...
+        futureSequence(psubsToUpdate) {
+          psubsByYear =>
+            val taxYear = psubsByYear._1
+            val claimAmount = claimAmountMinusDeductions(psubsByYear._2)
+
             taiService.updatePsubAmount(nino, taxYear, claimAmount)
         }
```
Comment (JoeSimmonds) [inline app/services/SubmissionService.scala]: or perhaps the whole thing can be reduced to something like :

```
claimYears =>
        Future.sequence(for {
          year <- claimYears
          psubs <- subscriptions.get(getTaxYear(year)).filter(_.nonEmpty)
        } yield taiService.updatePsubAmount(nino, getTaxYear(year), claimAmountMinusDeductions(psubs)))
```

Not sure if that's semantically the same.  Chat on Monday

File: app/services/SubmissionService.scala (hunk 1 above)
Comment (Pascoe121) [reply app/services/SubmissionService.scala]: implemented :)

File: app/services/SubmissionService.scala (hunk 2 above)
Comment (Pascoe121) [reply app/services/SubmissionService.scala]: Wouldn't work due to needing to submit claims in a procedural manner due to etag incrementation.

**Pull request merged**